pkg/tuned/change-profile.jsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "react",
+], function(React) {
+
+"use strict";
+
+/* Performance profile entry
+ * Expected props:
+ *  - name (key)
+ *  - recommended (boolean)
+ *  - active (boolean)
+ *  - selected (boolean)
+ *  - title
+ *  - description
+ *  - click (callback function)
+ */
+var TunedDialogProfile = React.createClass({
+    render: function() {
+        var classes = "list-group-item";
+        if (this.props.selected)
+            classes += " active";
+        var recommended;
+        if (this.props.recommended)
+            recommended = <span className="badge pull-right">recommended</span>;
+        return (
+            <div className={ classes } key={ this.props.name } onClick={ this.props.click }>
+                {recommended}
+                <p>{ this.props.title }</p>
+                <small>{ this.props.description }</small>
+            </div>
+        );
+    }
+});
+
+/* dialog body with list of performance profiles
+ * Expected props:
+ *  - active_profile (key of the active profile)
+ *  - change_selected callback, called with profile name each time the selected entry changes
+ *  - profiles (array of entries passed to TunedDialogProfile)
+ *    - name (string, key)
+ *    - recommended (boolean)
+ *    - active (boolean)
+ *    - title (string)
+ *    - description (string)
+ */
+var TunedDialogBody = React.createClass({
+    getInitialState: function() {
+        return {
+            selected_profile: this.props.active_profile,
+        };
+    },
+    componentDidMount: function() {
+        this.props.change_selected(this.props.active_profile);
+    },
+    handleProfileClick: function(profile) {
+        if (profile != this.state.selected_profile) {
+            this.setState( { selected_profile: profile } );
+            this.props.change_selected(profile);
```

Comment:
Why do we change the props and the state as a result of this change? Shouldn't it just track in state until the user actually clicks the button? Or are we using a different scheme here?